### ISSUE TITLE
Remove all code references to the mature-related models without removing the models themselves

### ISSUE THREAD

**Issue by AetherUnbound** [WordPress/openverse#2629]
## Description

Remove all code references to the old models (`MatureImage`, `MatureAudio`, `NsfwReport`, `NsfwReportAudio`) from the API, but leave the models in place for the time being. Instructions for how to perform this change are present in the implementation plan associated with this work: https://docs.openverse.org/projects/proposals/trust_and_safety/content_report_moderation/20230517-implementation_plan_mature_copy_change.html#code (step 7).

A deployment of the API will be necessary before handling downstream tickets.

## Additional context

Part of #383, blocked by #2627, #2628

**Comment by AetherUnbound** [comment]
Closing in favor of #3723 based on #3568.
